samza-api/src/main/java/org/apache/samza/operators/MessageStream.java

```diff
@@ -82,17 +82,19 @@
    * non-stream systems (e.g., an external database).
    *
    * @param sinkFn the function to send messages in this stream to an external system
+   * @return this {@link MessageStream} object
    */
-  void sink(SinkFunction<M> sinkFn);
+  MessageStream<M> sink(SinkFunction<? super M> sinkFn);
```

Comment:
I think it'll be safer and more clearly explainable to leave them as terminal. Users can still chain explicitly on the previous message stream if they wish to. Don't feel strongly about it, up to you.